packages/swing-store/src/snapStore.js

```diff
@@ -34,10 +36,14 @@ import { fsStreamReady } from '@agoric/internal/src/fs-stream.js';
  *   deleteVatSnapshots: (vatID: string) => void,
  *   deleteSnapshotByHash: (vatID: string, hash: string) => void,
  *   getSnapshotInfo: (vatID: string) => SnapshotInfo,
+ *   getExportRecords: (includeHistorical: boolean) => Iterable<[key: string, value: string]>,
+ *   getArtifactNames: (includeHistorical: boolean) => Iterable<string>,
+ *   getSnapshot: (vatID: string, endPos: number) => AsyncIterable<Uint8Array>,
+ *   importSnapshot: (artifactName: string, exporter: SwingStoreExporter, artifactMetadata: Map) => void,
```

Comment:
I find this a little too much of an abstraction leak to reveal the `exporter` and this metadata collection to `importSnapshot`. Why can't we feed it the parts relevant for the named artifact instead?

```suggestion
 *   importSnapshot: (artifactName: string, artifactMetadata: string, artifactDataStream: AsyncGenerator<Uint8Array>) => void,
```

Reply:
No, the way this is used it needs to be a random access data structure.  I think the problem is that there are methods on the snapStore and transcriptStore (of which this is one) that are actually internal to the swingStore implementation and shouldn't really be part of the public API.  However, the needs of testing push against that because tests need access to them too and the API itself isn't very public anyway.  One could imagine a complete refactoring of the swingStore API that was less explicit about the more specialized constituent stores, and that would probably be a cleaner design, but given that this really isn't a very public API I don't think the effort required for such refactoring would pay for itself soon and certainly not within our shipping time horizon.